app/services/github/contributions/collect.rb

```diff
@@ -3,44 +3,59 @@
 module Github
   module Contributions
     class Collect
-      Result = Struct.new(:uid, :rid, :pull_request)
+      Result = Struct.new(:uid, :rid, :pull_request_url, :created_at)
 
       def initialize(company:, client:)
         @company = company
         @client = client
       end
 
       def all
-        return [] if company.blank?
+        return [] if company.blank? or not repositories_wrapper.has_repositories? or not engineers_wrapper.has_engineers?
+        fetch_all
+      end
+
+      private
 
-        repositories.flat_map do |repository_id, repository_owner, repository_name|
-          client.pull_requests
-                .list(repository_owner, repository_name)
-                .map do |pull_request|
-                  uuid, = engineers.select { |_, username| username ==  pull_request.user.login }
-                                   .flatten
+      attr_reader :company, :client
 
-                  Result.new(uuid, repository_id, pull_request) if uuid.present?
-                end.compact
+      def fetch_all
+        begin
+          yesterday_date = 1.day.ago.strftime("%Y-%m-%d")
+          query = "created:#{yesterday_date} is:pr #{authors_query} #{repositories_query}"
+          fetch_pull_requests(search_query: query)
+        rescue => e
```

Comment:
avoid rescuing `Exceptions`, use at least `StandardError` you can understand why [here](https://www.honeybadger.io/blog/ruby-exception-vs-standarderror-whats-the-difference/)

Reply:
that's interesting :rocket: